src/biz/workspace/page_view.rs

```diff
@@ -0,0 +1,229 @@
+  let publish_view_ids: HashSet<String> = publish_view_ids
+    .into_iter()
+    .map(|id| id.to_string())
+    .collect();
+  let folder_view = FolderView {
+    view_id: view_id.to_string(),
+    name: view.name.clone(),
+    icon: view
+      .icon
+      .as_ref()
+      .map(|icon| to_dto_view_icon(icon.clone())),
+    is_space: view_is_space(&view),
+    is_private: false,
+    is_published: publish_view_ids.contains(view_id),
+    layout: to_view_layout(&view.layout),
+    created_at: DateTime::from_timestamp(view.created_at, 0).unwrap_or_default(),
+    last_edited_time: DateTime::from_timestamp(view.last_edited_time, 0).unwrap_or_default(),
+    extra: view.extra.as_ref().map(|e| parse_extra_field_as_json(e)),
+    children: vec![],
+  };
+  let page_collab_data = match view.layout {
+    collab_folder::ViewLayout::Document => {
+      get_page_collab_data_for_document(
+        collab_access_control_storage.clone(),
+        uid,
+        workspace_id,
+        view_id,
+      )
+      .await
+    },
+    collab_folder::ViewLayout::Grid
+    | collab_folder::ViewLayout::Board
+    | collab_folder::ViewLayout::Calendar => {
+      get_page_collab_data_for_database(
+        pg_pool,
+        collab_access_control_storage.clone(),
+        uid,
+        workspace_id,
+        view_id,
+      )
+      .await
+    },
+    collab_folder::ViewLayout::Chat => Err(AppResponseError::new(
+      ErrorCode::InvalidRequest,
+      "Page view for AI chat is not supported at the moment",
+    )),
+  }?;
+
+  let page_collab = PageCollab {
+    view: folder_view,
+    data: page_collab_data,
+    owner,
+    last_editor,
+  };
+
+  Ok(page_collab)
+}
+
+async fn get_page_collab_data_for_database(
+  pg_pool: &PgPool,
+  collab_access_control_storage: Arc<CollabAccessControlStorage>,
+  uid: i64,
+  workspace_id: Uuid,
+  view_id: &str,
+) -> Result<PageCollabData, AppResponseError> {
+  let ws_db_oid = select_workspace_database_oid(pg_pool, &workspace_id).await?;
+  let ws_db = get_latest_collab_encoded(
+    collab_access_control_storage.clone(),
+    GetCollabOrigin::User { uid },
+    &workspace_id.to_string(),
+    &ws_db_oid,
+    CollabType::WorkspaceDatabase,
+  )
+  .await?;
+  let mut ws_db_collab = collab_from_doc_state(ws_db.doc_state.to_vec(), &ws_db_oid)?;
+  let ws_db_body = WorkspaceDatabaseBody::open(&mut ws_db_collab);
+  let txn = ws_db_collab.transact();
+  let db_oid = ws_db_body
+    .get_database_meta_with_view_id(&txn, view_id)
+    .ok_or(AppResponseError::new(
+      ErrorCode::NoRequiredData,
+      format!("Database view {} not found", view_id),
+    ))?
+    .database_id;
+  let db = get_latest_collab_encoded(
+    collab_access_control_storage.clone(),
+    GetCollabOrigin::User { uid },
+    &workspace_id.to_string(),
+    &db_oid,
+    CollabType::Database,
+  )
+  .await?;
+  let db_collab = Collab::new_with_source(
+    CollabOrigin::Server,
+    &db_oid,
+    DataSource::DocStateV1(db.doc_state.to_vec()),
```

Comment:
can use `db.into()`.

Reply:
`db.into()` would cause `db` to be moved, which makes it not possible to call db.doc_state later.